draft-ZIP-0226.rst

```diff
@@ -0,0 +1,303 @@
+
+This, however, brings some issues when it comes to adding multiple asset types, as the output note of the split Actions *cannot* be of *any* asset type, it must be enforced to be an actual output of a GroupHash computation (in fact we want it to be of the same type as the original input note, but the binding signature takes care that the proper balancing is performed). If not, then the prover could essentially input a multiple (or linear combination of) an existing type, with the goal to attack the network by overflowing the ZEC value balance and hence counterfeiting ZEC funds.
+
+In order to prevent this, we make some modifications to the circuit. Specifically we remove the dummy note functionality for custom assets and we enforce that *every* input note to an ZSA Action must be proven to exist in the set of note commitments in the Merkle Tree. We then enforce this real note to be “unspendable” in the sense that its value
+will be zeroed in split Actions and the nullifier will be randomized, making the note not spendable in the specific Action. Then, the proof itself ensures that the output note is of the same type as the input note. In the circuit, the split note functionality will be activated by a boolean private input to the proof.
+
+Note that this is enough to create a chain of induction that ensures that all output notes of a transfer are actual outputs of a GroupHash, preventing any malleability attacks, as they originate in the Issuance protocol, which is publicly verified. Furthermore, we do not care about whether the note is owned by the sender, or whether it was nullified before. Wallets and other clients have a choice to make to ensure the asset type is the preserved for the output note of a split Action, for the value balance verification:
+
+1. The split input note could be the same note as the original (non-split) Action, 
+2. The split input note could be a different unspent note of the same type (note that the note will not actually be spent)
+3. The split input note could be an already spent note of the same type (note that by zeroing the value in the circuit, we prevent double spending)
+
+The specific circuit changes are presented below.
+
+Circuit Statement
+=================
+
+The advantage of the design described above, with respect to the circuit statement, is that every *ZSA Action statement* is kept closely similar to the Orchard Action statement [#protocol-actionstatement]_, except for a few additions that ensure the security of the asset type system.
+
+**Asset Type Equality:** the following constraints must be added to ensure that the input and output note are of the
+same type:
+
+- The asset type, :math:`\mathsf{type_\mathbb{P}}`, for the note is witnessed once, as an auxiliary input.
+- The witnessed asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the old note commitment input.
+- The witnessed asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the new note commitment input.
```

Comment:
```suggestion
- The Asset type, :math:`\mathsf{type_\mathbb{P}}`, for the note is witnessed once, as an auxiliary input.
- The witnessed Asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the old note commitment input.
- The witnessed Asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the new note commitment input.
```

Reply:
The is too vague. Need a precise specification of how the note commitment is changed.

Reply:
Agreed, working on that.